packages/documentation-v7/src/stories/components/range/range.stories.ts

```diff
@@ -0,0 +1,288 @@
+      control: {
+        type: 'radio',
+        labels: {
+          'null': 'Default',
+          'is-valid': 'Valid',
+          'is-invalid': 'Invalid',
+        },
+      },
+      options: ['null', 'is-valid', 'is-invalid'],
+      table: {
+        category: 'States',
+      },
+    },
+  },
+};
+
+export default meta;
+
+type Story = StoryObj;
+
+function render(args: Args, context: StoryContext) {
+  const [_, updateArgs] = useArgs();
+
+  const id = `${context.viewMode}_${context.name.replace(/\s/g, '-')}_ExampleRange`;
+  const classes = ['form-range', args.validation].filter(c => c && c !== 'null').join(' ');
+
+  const useAriaLabel = args.hiddenLabel;
+  const label: TemplateResult | null = !useAriaLabel
+    ? html`
+        <label class="form-label" for="${id}">${args.label}</label>
+      `
+    : null;
+
+  const contextual: (TemplateResult | null)[] = [
+    args.validation === 'is-valid'
+      ? html`
+          <p class="valid-feedback">Ggranda sukceso!</p>
+        `
+      : null,
+    args.validation === 'is-invalid'
+      ? html`
+          <p class="invalid-feedback">Eraro okazis!</p>
+        `
+      : null,
+  ];
+
+  const control: TemplateResult = html`
+    <input
+      id="${id}"
+      class="${classes}"
+      type="range"
+      value="${args.value}"
+      .value="${args.value}"
+      min="${args.useBoundaries ? args.min : nothing}"
+      max="${args.useBoundaries ? args.max : nothing}"
+      step="${args.useBoundaries ? args.step : nothing}"
+      ?disabled="${args.disabled}"
+      aria-label="${useAriaLabel ? args.label : nothing}"
+      ?aria-invalid="${VALIDATION_STATE_MAP[args.validation]}"
+      @input="${(e: MouseEvent) => updateArgs({ value: (e.target as HTMLInputElement).value })}"
+      @keyup="${(e: KeyboardEvent) => {
+        if (ARROW_KEYS.includes(e.key)) {
+          if (document.activeElement === e.target) {
+            setTimeout(() => {
+              const element: HTMLInputElement | null = document.querySelector(`#${id}`);
+              if (element) element.focus();
+            }, 25);
+          }
+        }
+      }}"
+    />
+  `;
+
+  let valueElement: TemplateResult | TemplateResult[] | null = null;
+
+  if (args.showValue === 'text') {
+    valueElement = html`
+      <p class="form-text">${args.value}</p>
+    `;
+  } else if (args.showValue === 'input') {
+    const inputId = `${context.viewMode}_${context.name.replace(/\s/g, '-')}_ExampleRangeInput`;
+
+    valueElement = [
+      html`
+        <label class="form-label visually-hidden" for="${inputId}">Range controller</label>
+      `,
+      html`
+        <input
+          id="${inputId}"
+          class="form-control mw-giant"
+          type="text"
+          inputmode="decimal"
+          value="${args.value}"
+          .value="${parseFloat(args.value)}"
+          ?disabled="${args.disabled}"
+          @input="${(e: Event) => updateArgs({ value: (e.target as HTMLInputElement).value })}"
```

Comment:
The `@input` binding leads to a `NaN` output in the input if a user wants to enter a negative number.
We should probably change this to an `@change` binding.

Reproduction:
- Set `Boundaries` control to `true`
- Set `Minium Value` to a negative number
- Set `Show Value` to `input`
- Try to enter a negative number in the input

Reply:
The parseFloat was the issue actually, but thanks for the catch!